api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -1848,6 +1853,48 @@ public void savePatient_shouldUpdateAnExistingPatient() throws Exception {
 		assertTrue("The gender should be new", patient2.getGender().equals("F"));
 	}
 	
+	@Test
+	public void savePatient_shouldFailSaveIfPatientsWithSameIdentifierAreSaved() {
+	    isSavePatientThreadsafe = new AtomicBoolean(false);
```

Comment:
Just one opinion,
`isSavePatientThreadsafe` would be changed to `true` by default or `isSavePatientThreadUnSafe = false` by default will good for the naming. Because at the beginning all threads are in safe state and it will become the critical stage while you are performing the saving method.

Reply:
@samuelmale did you see the above?

Reply:
@samuelmale did you see the above?